assets/scripts/fe/src/app/components/users/setting/setting.component.ts

```
@@ -27,10 +40,24 @@ export class SettingComponent implements OnInit {
 
   async ngOnInit() {
     // initialize the form.
+    this.switch_expression = null
     this.form = new UserForm(new User);
 
     this.nav.setNav('Profile', true);
 
+    this.edit_password_form = new EditPasswordForm(new EditPasswordModel)
+    this.add_password_form = new AddPasswordForm(new AddPasswordModel)
+
+    // check if user has a usable password
+    this.userservice.hasPass().subscribe(
+      data => {
+        this.checkPass = data
+      },
+      error => {
+        console.log(error)
```

do not hide error message unless you have a valid reason